### PR TITLE
split rubocop files for dependency on manageiq

### PR DESCRIPTION
**Pull request by durandom**
* changes rubocop config to point to manageiq repo instead of manageiq/guides
* don't maintain our own copy of rubocop_cc.yml for codeclimate, but use their support for [external configurations](https://codeclimate.com/changelog/582495c32c33066f1b00191d) 

following along https://github.com/ManageIQ/manageiq-providers-amazon/pull/90

### CODE REVIEW

**Comment by djberg96** [comment]
@durandom Quick question, will spec/manageiq/.rubocop.yml need any adjustment?

**Comment by durandom** [comment]
@djberg96 `spec/manageiq/.rubocop.yml` is not used at all. All under `spec/manageiq` is just the dummy app to run specs and pulls in dependencies from the `spec/manageiq/Gemfile`

**Comment by djberg96** [comment]
@durandom Ok, thanks for the clarification.

**Comment by durandom** [comment]
@bronaghs can you have a look and merge?

File: .rubocop_local.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+GlobalVars:
+  AllowedVariables:
+  # Loggers
+  - $log
```
Comment (bronaghs) [inline .rubocop_local.yml]: @durandom - looks like the azure_log be added here too, right?

**Comment by durandom** [comment]
@bronaghs $azure_log is in. Good catch.

**Pull request merged**